app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -552,17 +729,90 @@ export class MetamaskNotificationsController extends BaseController<
       await this.#storage.setNotificationStorage(JSON.stringify(userStorage));
 
       // Update the state of the controller
+      this.setMetamaskNotificationsFeatureSeen();
       this.setFeatureAnnouncementsEnabled(true);
-      this.setMetamaskNotificationsEnabled(true);
       this.setSnapNotificationsEnabled(true);
+      this.setMetamaskNotificationsEnabled(true);
+      this.#setIsUpdatingMetamaskNotifications(false);
 
       return userStorage;
     } catch (err) {
+      this.#setIsUpdatingMetamaskNotifications(false);
       log.error('Failed to create On Chain triggers', err);
       throw new Error('Failed to create On Chain triggers');
     }
   }
 
+  /**
+   * Enables all MetaMask notifications for the user.
+   * This method performs several key operations:
+   * 1. Validates the storage key and bearer token necessary for authentication.
+   * 2. Retrieves all linked accounts.
+   * 3. Ensures profile syncing is enabled if the storage key is missing.
+   * 4. Enables on-chain triggers for each account.
+   * 5. Sets the global notification settings for MetaMask, feature announcements, and Snap notifications to true.
+   *
+   * @throws {Error} If there is an error during the process of enabling notifications.
+   */
+  public async enableMetamaskNotifications() {
+    try {
+      this.#setIsUpdatingMetamaskNotifications(true);
+
+      await this.#performEnableProfileSyncing();
+
+      const { accounts } = await this.#accounts.listAccounts();
+
+      // For each account enable the triggers
+      await this.updateOnChainTriggersByAccount(accounts);
```

Comment:
As discussed/demoed, this is potentially a bug.

It might be fine to create triggers (if they are missing), but we do not want to delete triggers if you are disabling notifications.